src/main/java/com/kodilla/ecommercee/controller/UserController.java

```diff
@@ -1,27 +1,48 @@
 package com.kodilla.ecommercee.controller;
 
-import com.kodilla.ecommercee.domain.GenericEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.domain.UserDto;
+import com.kodilla.ecommercee.exception.UserNotFoundException;
+import com.kodilla.ecommercee.mapper.UserMapper;
+import com.kodilla.ecommercee.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/v1/user")
 public class UserController {
 
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private UserMapper userMapper;
+
     @PostMapping(value = "createUser")
-    public GenericEntity createUser() {
-        return new GenericEntity("user_1");
+    public UserDto createUser(@RequestParam String username, @RequestParam String eMail, @RequestParam String address) {
+
+        User user = new User(username, eMail, address);
+        return userMapper.mapToUserDto(userService.saveUser(user));
     }
 
     @PutMapping(value = "blockUser")
-    public GenericEntity blockUser(Long userId) {
-        return new GenericEntity("user_blocked");
+    public UserDto blockUser(@RequestParam Long userId) {
+
+        try {
+            return userMapper.mapToUserDto(userService.blockUser(userId));
+        } catch(UserNotFoundException e) {
+            return e.getNullUser();
+        }
+
     }
 
     @PostMapping(value = "createUserKey")
-    public String createUserKey(String userName, String userKey) {
-        return "user_key";
+    public String createUserKey(@RequestParam Long userId) {
```

Comment:
Klucz powinien być generowany dla użytkownika po podaniu poprawnych danych użytkownika (username i password może). Generując klucz mając tylko id to możnaby sobie generować klucze innych użytkowników

Reply:
Myślałem tutaj o sytuacji, że będzie oddzielna klasa do validacji usera za pomocą hasła itd (tak jak rozmawialiśmy na jednym ze spotkań), i to ona po validacji przekazywała by do metody createUserKey id.
Jednak po przemyśleniu jeszcze raz coś nie gra, bo jak jakaś klasa miałaby przekazać parametr do requesta HTTP... 
Poprawię :)